language-design.md

```diff
@@ -74,7 +120,17 @@ implementation choices from users?
 
 **Response**
 
+Implementing an internal DSL will give it the feeling of the host language, allowing more
+familiarity with the syntax, but it won't give the developers as much control over what the
+users can do. 
+The choice of interpreter versus compiler will affect the build process, or whether a user
+has to compile before running a program.
 
+In general, we can't always hide our implementation choices. Obviously, whether you must
+compile or not should be known by the user.
+When it is possible to hide our implementation choices, it makes it easier to change them
+later on without changing the user's experience, so choices like using an interpreter or a 
+compiler should be made carefully as they are probably going to be permanent.
```

Comment:
I agree that hiding implementation allows for future changes to occur more smoothly.

Are there ever any benefits to exposing implementation.